File: update_context.py
import json
import datetime

# Read the file
with open('.gemini_agent_context.json', 'r') as f:
    data = json.load(f)

# Update root keys
data['last_agent'] = 'qa-agent'
data['next_agent'] = 'dev-agent'

# Create/Update round-4 qa-agent entry
if 'round-4' not in data:
    data['round-4'] = {}

data['round-4']['qa-agent'] = {
    "timestamp": datetime.datetime.now(datetime.timezone.utc).isoformat(),
    "qa_verdict": "FAIL",
    "failure_details": "Playwright E2E tests failed with 14 failures. While 'polish-with-ai-button' tests passed, systemic regressions were observed across several modules (AI Chatbot, Schedule Navigation, Global Search, Analytics). Errors included 429 Too Many Requests, timeouts, and element visibility issues indicating broken data-seeding or race conditions despite the full project re-seed.",
    "expected_output": {
        "commands": ["npx playwright test"],
        "success_indicators": ["Zero failing E2E tests.", "Clean console output (no hydration warnings or 4xx/5xx)."]
    }
}

# Write back
with open('.gemini_agent_context.json', 'w') as f:
    json.dump(data, f, indent=2)

File: update_context.js
const fs = require('fs');

const contextData = JSON.parse(fs.readFileSync('.gemini_agent_context.json', 'utf8'));

contextData.last_agent = "dev-agent";
contextData.next_agent = "review-agent";
contextData.last_updated_at = new Date().toISOString();

contextData["round-2"] = {
  "dev-agent": {
    "timestamp": new Date().toISOString(),
    "files_modified": [
      "src/__tests__/e2e/analytics.spec.ts",
      "src/__tests__/e2e/chat.spec.ts",
      "src/__tests__/e2e/schedule-navigation.spec.ts",
      "src/__tests__/e2e/ticket-538-roles-cleanup.spec.ts",
      "scripts/seed-e2e-schedule.ts",
      "src/auth.ts",
      "src/app/schedule/page.tsx",
      "src/__tests__/unit/video-lifecycle.test.ts"
    ],
    "expected_output": {
      "status": "PASS",
      "details": "All Playwright E2E tests, including analytics, chat, schedule-navigation, and roles, now pass. Fixed seed scripts causing DB constraint failures, mocked chat API to fix slow AI timeouts, fixed hydration mismatch with isMounted, and improved test setups. Run npx playwright test, npx tsc --noEmit, and npm run build to verify."
    }
  }
};

fs.writeFileSync('.gemini_agent_context.json', JSON.stringify(contextData, null, 2));
